edweiss-firebase/functions/src/utils/firebase.ts

```diff
@@ -16,7 +17,37 @@ export function onAuthentifiedCall<Args, Result, Error>(
 		if (!request.auth?.uid)
 			return NOT_AUTHENTIFIED;
 
-		return await callback(request.auth.uid, request.data);
+		try {
+			return await callback(request.auth.uid, request.data);
+		} catch (e: FailedCallResult<Error> | any) {
+			if (e.status == 0)
+				return e;
+
+			return INTERNAL_ERROR;
+		}
+	});
+
+	return { signature, handler };
+}
+
+export function onSanitizedCall<Args extends object, Result, Error>(
```

Comment:
What does this function achieve? I saw that you replaced an onAuthentifiedCall with it earlier (createQuiz function).

Reply:
Hello,
This achieves practically the same thing as `onAuthentifiedCall`, it simply checks if the argument predicates are satisfied before the callback.
But you're right, I forgot to document it.

Reply:
I see. Good job as always, this is an absolutely bonkers amount of work! :+1: